tokio/src/runtime/time/entry.rs

```diff
@@ -300,11 +300,17 @@ pub(crate) struct TimerEntry {
     inner: StdUnsafeCell<TimerShared>,
     /// Initial deadline for the timer. This is used to register on the first
     /// poll, as we can't register prior to being pinned.
-    initial_deadline: Option<Instant>,
+    deadline: Deadline,
```

Comment:
Why not just an instant and a boolean?

Reply:
I did consider that at first. I felt using an enum was clearer. Happy to go with either

Reply:
Reading over it again, I still think that a boolean is clearer. I would prefer if you switch it to a boolean.

Reply:
Done